### ISSUE TITLE
Existing file check algorithm uses SHA1 encryption

### ISSUE THREAD

**Issue by maskaravivek** [commons-app/apps-android-commons#384]
`ShareActivity` uses `getSHA1` method of `Utils` class to generate SHA1 of file from input stream. Given the vulnerabilities exposed with SHA1(not sure how critical it is in our use case) we should consider moving to SHA2 for our file matching algorithms. Both the server and client will have to incorporate these changes if we go forward with it.

**Comment by whym** [comment]
Yes, and we will have to wait for MediaWiki to migrate: https://phabricator.wikimedia.org/T158986 (and when MediaWiki migrates to another hash, we would like to be not left behind).

**Comment by chrisdebian** [comment]
@maskaravivek @whym Hi, both.

It sounds like this has probably been overtaken by events; can we confirm whether SHA2 is now being used, so we can potentially close this?

Thanks,

Chris.
